graylog2-server/src/main/java/org/graylog2/indexer/rotation/strategies/SmartRotationStrategy.java

```diff
@@ -50,14 +66,83 @@ public RotationStrategyConfig defaultConfiguration() {
         return SmartRotationStrategyConfig.builder().build();
     }
 
-    @Nullable
     @Override
+    @Nonnull
     protected Result shouldRotate(final String index, IndexSet indexSet) {
-        return null;
+        final DateTime creationDate = indices.indexCreationDate(index).orElseThrow(()-> new IllegalStateException("No index creation date"));
+        final Long sizeInBytes = indices.getStoreSizeInBytes(index).orElseThrow(() -> new IllegalStateException("No index size"));
+
+        Period leeWay;
+        if (indexSet.getConfig().rotationStrategy() instanceof SmartRotationStrategyConfig rotationConfig) {
+            leeWay = rotationConfig.indexLifetimeHard().minus(rotationConfig.indexLifetimeSoft());
+        } else {
+            throw new IllegalStateException(f("Unsupported RotationStrategyConfig type <%s>", indexSet.getConfig().rotationStrategy()));
+        }
```

Comment:
Do we need this check? It shouldn't be possible to create an instance of `SmartRotationStrategy` with a mismatched config.
I do see that other rotation strategy classes have the same check, so we should do it just for consistency.

Reply:
I think it doesn't hurt to have it. What else would you do in this case? Just return silently and do nothing?